x/supplier/types/message_submit_proof.go

```diff
@@ -40,10 +40,38 @@ func (msg *MsgSubmitProof) GetSignBytes() []byte {
 	return sdk.MustSortJSON(bz)
 }
 
+// ValidateBasic ensures that the bech32 address strings for the supplier and
+// application addresses are valid and that the proof and service ID are not empty.
+//
+// TODO_TECHDEBT: Call `msg.GetSessionHeader().ValidateBasic()` once its implemented
 func (msg *MsgSubmitProof) ValidateBasic() error {
-	_, err := sdk.AccAddressFromBech32(msg.SupplierAddress)
+	_, err := sdk.AccAddressFromBech32(msg.GetSupplierAddress())
 	if err != nil {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid supplierAddress address (%s)", err)
+		return sdkerrors.ErrInvalidAddress.Wrapf(
```

Comment:
```suggestion
		return ErrSupplierInvalidAddress.Wrapf(
```
I think you should define this type something like `invalid supplier address`

Reply:
I created #348 to capture the higher-level concerns that I see as in play here.

In this particular case, my rationale for using `sdkerrors.ErrInvalidAddress` over any of our own error variables is that the usage here is identical to that of the what I interpreted as its intended usage, as indicated by its actual usage in cosmos-sdk code (i.e. in cases where `sdk.AccAddressFromBech32()` returns an error in `#ValidateBasic()` methods).

Reply:
I could imagine a middle-of-the-road alternative as us defining an alias to that error, such that we reference the alias instead. However, this begs the question, "in which module should that error variable be defined?"